source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -589,7 +589,7 @@ static int s_increment_read_window(struct aws_channel_handler *handler, struct a
         aws_channel_slot_increment_read_window(slot, window_update_size);
     }
 
-    if (secure_transport_handler->negotiation_finished) {
+    if (secure_transport_handler->negotiation_finished && secure_transport_handler->read_task.node.next) {
```

Comment:
Are you certain you didn't mean this?

! secure_transport_handler->read_task.node.next

Reply:
Wow it's amazing this worked

Reply:
heh... you just lucked out that you never got a partial packet.